files/build_templates/init_cfg.json.j2

```diff
@@ -26,6 +26,33 @@
             "FLEX_COUNTER_STATUS": "enable",
             "FLEX_COUNTER_DELAY_STATUS": "true",
             "POLL_INTERVAL": "10000"
+        },
+        "PORT": {
+            "FLEX_COUNTER_STATUS": "enable"
+        },
+        "RIF": {
+            "FLEX_COUNTER_STATUS": "enable"
+        },
+        "QUEUE": {
+            "FLEX_COUNTER_STATUS": "enable"
+        },
+        "PFCWD": {
+            "FLEX_COUNTER_STATUS": "enable"
+        },
+        "PG_WATERMARK": {
+            "FLEX_COUNTER_STATUS": "enable"
+        },
+        "PG_DROP": {
+            "FLEX_COUNTER_STATUS": "enable"
+        },
+        "QUEUE_WATERMARK": {
+            "FLEX_COUNTER_STATUS": "enable"
+        },
+        "BUFFER_POOL_WATERMARK": {
+            "FLEX_COUNTER_STATUS": "enable"
+        },
+        "PORT_BUFFER_DROP": {
+            "FLEX_COUNTER_STATUS": "enable"
```

Comment:
Let's treat this as a new feature, feature enabled by changining config_db.json. Please keep all the exsiting init_cfg.json / minigraph.py behavior unchanged, we are moving away from minigraph based config flow.

It is acceptable to define new field in config_db.json schema, indicating static/dynamic flex_counter mode, if config_db.json missing the new field, using default value (dynamic), and behavior is current behavior. So we also maintain backward compatible.

Reply:
@qiluo-msft Why this change is not backward compatible? Not having these entries in init_cfg and not adding them in runtime on the other hand is backward incompatible (no counters enabled after upgrade). Why defining a new field? How moving away from minigraph based flow is related?

Reply:
@qiluo-msft This is the flow of `config reload`:

```
admin@sonic:~$ sudo config reload -y /home/admin/config_db.json
Disabling container monitoring ...
Stopping SONiC target ...
Running command: /usr/local/bin/sonic-cfggen -j /etc/sonic/init_cfg.json -j /home/admin/config_db.json --write-to-db
Running command: /usr/local/bin/db_migrator.py -o migrate
Running command: /usr/local/bin/sonic-cfggen -d -y /etc/sonic/sonic_version.yml -t /usr/share/sonic/templates/sonic-environment.j2,/etc/sonic/sonic-environment
Restarting SONiC target ...
Enabling container monitoring ...
Reloading Monit configuration ...
Reinitializing monit daemon
```

The config load 1) merges config with init_cfg.json 2) runs db_migrator to handle breaking changes.

1. If `/home/admin/config_db.json` does not have any FLEX_COUNTER configuration it will be taken from `init_cfg.json` so that counters are going to be enabled. No breaking changes, same config_db.json would produce exactly the same FC configuration via `enable_counters.py` script in the old image.
2. If `/home/admin/config_db.json` has FLEX_COUNTER configuration (e.g. some counters were disabled before) it won't be overwritten by `init_cfg.json`. No breaking change.

Reply:
@stepanblyschak As we discussed, share the command we are using to config device `sudo config load_minigraph --override_config -y`

Reply:
@qiluo-msft I reviewed the command `config load_minigraph --override_config -y`. It calls `config override-config-table /etc/sonic/golden_config_db.json` command which for every table in `/etc/sonic/golden_config_db.json` removes table from CONFIG_DB and replaces with the content from the file.

I find that the behavior of `config override-config-table` is inconsistent with `config reload`. In order to handle breaking changes correctly it should do the same as `config reload`:
1.  Produce the config `/usr/local/bin/sonic-cfggen -j /etc/sonic/init_cfg.json -j /etc/sonic/golden_config_db.json`
2. Run the config replacement logic
3. Run `/usr/local/bin/db_migrator.py -o migrate`

So I believe this is a bug in `config override-config-table` command.

Reply:
config override-config-table will not run db_migrator since the config source is gonna provide bt other source of truth.

Reply:
@wen587 I don't follow, could you please clarify? Why is `override-config-table` different from `reload`?

Reply:
You can see the db_migrator is executed earlier than override: https://github.com/sonic-net/sonic-utilities/blob/master/config/main.py#L2086-L2103
That's because we want the golden config to be the single source of truth for configuration. It will help us to incremental configthe device(such as using GCU:generic config updater)  if there is any issue with the device.